pkg/daemon/update.go

```diff
@@ -444,6 +451,21 @@ func (mcDiff *machineConfigDiff) isEmpty() bool {
 	return reflect.DeepEqual(mcDiff, &emptyDiff)
 }
 
+// osChangesString generates a human-readable set of changes from the diff
+func (mcDiff *machineConfigDiff) osChangesString() string {
+	changes := []string{}
+	if mcDiff.osUpdate {
+		changes = append(changes, "Upgrading OS")
```

Comment:
only thing is we lost the OSImageURL in the event from line 1315 below. can we re-add? otherwise lgtm

Reply:
[Per above](https://github.com/openshift/machine-config-operator/pull/1977#discussion_r464638935) though with these events it's mostly about timing, so removing the URL was intentional.  

I think this really leads into a larger ergnonomic issue in that we should really be including something like a version number in MachineConfigs, and perhaps even include separate versions for our "base" MCs versus anything user provided.

For example for status we'd report `All nodes are updated to config v7 (rendered-<md5here>)` or so where we increment that counter each time we generate a new rendered MC.

Similarly we should be showing the ostree version number and not just the sha256.

I can re-add it if you really want, I just personally didn't find it useful - the more interesting information is the MachineConfig names which already contains the osimageurl.

Reply:
OK the next problem with doing what you're asking is `machineConfigDiff` doesn't have the target...and it leads into the question of why are we special casing osImageURL in the event?  Why not also show the kargs that changed?  Why not try to show which systemd units?  It wouldn't scale - I think these events should just be about timing.

Reply:
Or to rephrase, the data in this PR is *more useful* because we're seeing if there are karg/unit changes too which we didn't have before - if we care to look up exactly what those are, we can look at the rendered config.

Adding osImageURL would make the event much more noisy when we don't need it.

Reply:
fair argument, we have other sources to look at what osImageURL is getting used like looking at MCD log or rendered MC. My main motivation of adding osImageURL is that this is the one which we pull in to perform upgrade, extensions and kernel switch.

Reply:
OK updated with a compromise - we keep the old event with the osimageurl and only emit it if we're pulling the oscontainer.

Reply:
Im cool with the compromise :)

Reply:
im ok with the compromise :)

Reply:
apparently im doubly ok with it :laughing: